models/MigrationService.cfc

```diff
@@ -35,6 +96,28 @@ component singleton accessors="true" {
 		return this;
 	}
 
+	public MigrationService function seed() {
```

Comment:
Said a different way:

If data is required for your application to work correctly, it should be done in a migration, not a seed.  A seed is a development tool for adding data quickly to your database.

Example:

In CFCasts we recently migrated the `language` field from `Series` to `Video`.  The migration that added the column to `Video` included a query to update `Video.language` = `Series.language` as the default.  This would not be a seed because the application would not work without that step.

If I was onboarding a new developer and they wanted to have CFCasts running locally with a dozen series and hundreds of videos, a seed would be perfect.  They could even run it twice if they wanted twice the records.

Reply:
One more note, Laravel actually prevents running seeders in production without a flag.  I'm not sure if we can do the same, but it would help show the difference between a migration and a seeder.

Reply:
I still don't think that seeders should log as migrations.

1. They don't have `up` and `down`.
2. They are optional.
3. It would be confusing to have one environment that has more logged migrations than another when they are on the same commit.

Your test case below is, to me, a perfect example of what I'm talking about.

```
component {

	function up( schema, query ) {
		var mockData = new testbox.system.modules.mockdatacfc.models.MockData();

		var testUsers = mockData.mock(
			argumentCollection = {
				"$num"     : 20,
				"email"    : "email",
				"password" : "word"
			}
		);

		query.table( "users" ).insert( testUsers );
	}

}
```

1. This only runs - it doesn't go `up` and `down`.
2. It is not required for the app to run.  If it was, it wouldn't be using `mockData`.
3. Running this multiple times doesn't change how the app runs, it just gives me more development data.

I think Seeders need to be a separate concept from Migrations.

Reply:
That's fair.  Let me work on simplifying the seeders a bit and adding some restrictions on preventing them from running in certain environments.

Reply:
I agree with not having it as part of a migration process.  In reality its a separate and optional process where I see a Command to be separate for seeding.  If you send it plain, then run all seeders, or optionally choose a specific seeder to run.

```
migrations seed

migrations seed users
```

Reply:
I think having a `--seed` argument to some migrate commands like `fresh` make sense as well as having a dedicated `seed` command.

One question, right now the commands are under a `migrate` namespace.  Would a `seed` command be under this namespace (`migrate seed`) or under its own namespace (`seed`)?

Reply:
I think `--seed` should be a flag on the `migrate` command, but I could definitely see the usefulness of a separate command.  Once we get this put to bed, I'll work on a PR for `commandbox-migrations` to incorporate the new command

Reply:
I've decoupled this from the migrations in commit 33e6313  and now allow for a component name filter to be passed in to the `seed` function